src/gui/PropertyModel.cpp

```diff
@@ -0,0 +1,64 @@
+#include <PropertyModel.h>
+#include <iostream>
+PropertyModel::PropertyModel(QObject* parent) : QAbstractTableModel(parent)
+{
+
+}
+
+PropertyModel::~PropertyModel()
+{
+
+}
+
+int PropertyModel::rowCount(const QModelIndex& ) const
+{
+    return properties.size();
+}
+
+int PropertyModel::columnCount(const QModelIndex& ) const
+{
+    return 2;
+}
+
+QVariant PropertyModel::data(const QModelIndex& index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        if (index.column() == 0)
+            return properties[index.row()];
+        else if (index.column() == 1)
+            return propertySource->property(properties[index.row()].toStdString().c_str());
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags PropertyModel::flags(const QModelIndex& index) const
+{
+    if (index.column() == 1)
+        return QAbstractTableModel::flags(index) | Qt::ItemIsEditable;
+
+    return QAbstractTableModel::flags(index);
+}
+
+void PropertyModel::setPropertySource(QObject* propertySource_)
+{
+    propertySource = propertySource_;
+
+    if (!propertySource)
+        return;
+
+    auto prop = propertySource->property("exposedProperties");
+    if (prop.isNull() || !prop.isValid())
+        return;
+
+    for (const auto& p: prop.toStringList())
+        properties.push_back(p);
+}
+
+bool PropertyModel::setData(const QModelIndex& index, const QVariant& value, int role)
+{
+    if (role == Qt::EditRole)
```

Comment:
What about:

```c++
return (role == Qt::EditRole)
        ? propertySource->setProperty(properties[index.row()].toStdString().c_str(), value)
        : true;
```

Reply:
It's a matter of preference again. I prefer if version. Ternary operator is less readable for me.